scripts/release/generate_history_notes.py

```diff
@@ -0,0 +1,177 @@
+    history = '**{}**\n\n'.format(component)
+    for note in history_notes[component]:
+        history += '* {}\n'.format(note)
+    history += '\n'
+    return history
+
+
+def construct_core_history(component: str):
+    history = ''
+    for note in history_notes[component]:
+        history += '* {}\n'.format(note)
+    history += '\n'
+    return history
+
+
+def get_release_head() -> str:
+    response = requests.get(release_head_url)
+    if response.status_code != 200:
+        raise Exception("Request to {} failed with {}".format(release_head_url, response.status_code))
+
+    release_head_obj = json.loads(response.content.decode('utf-8'))
+    release_head_sha = release_head_obj['object']['sha']
+    return release_head_sha
+
+
+def get_commits():
+    today = datetime.date.today()
+    start_date = today - datetime.timedelta(days=days_before)
+    start_date_str = start_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+    dev_since_url = '{}&since={}'.format(dev_commits_url, start_date_str)
+    response = requests.get(dev_since_url)
+    if response.status_code != 200:
+        raise Exception("Request to {} failed with {}".format(dev_since_url, response.status_code))
+    dev_commits = json.loads(response.content.decode('utf-8'))
+    return dev_commits
+
+
+def get_prs_for_commit(commit: str):
+    headers = {'Accept': 'application/vnd.github.groot-preview+json'}
+    url = commit_pr_url.replace('commit_id', commit)
+    response = requests.get(url, headers=headers)
+    if response.status_code != 200:
+        raise Exception("Request to {} failed with {}".format(url, response.status_code))
+    prs = json.loads(response.content.decode('utf-8'))
+    return prs
+
+
+def process_pr(pr):
+    lines = [pr['title']]
+    body = pr['body']
+    search_result = re.search(r'\*\*History Notes:\*\*(.*)---', body, flags=re.DOTALL)
+    if search_result is None:
+        search_result = re.search(r'\*\*History Notes:\*\*(.*)', body, flags=re.DOTALL)
+        if search_result is None:
+            return
+    body = search_result.group(1)
+    lines.extend(body.splitlines())
+    process_lines(lines)
+
+
+def process_commit(commit):
+    lines = commit['commit']['message'].splitlines()
+    process_lines(lines)
+
+
+def process_lines(lines: [str]):
+    # do not put note of hotfix here since it's for last release
+    if re.search('hotfix', lines[0], re.IGNORECASE):
+        return
+    note_in_desc = False
+    for desc in lines[1:]:
+        component, note = parse_message(desc)
+        if component is not None:
+            note_in_desc = True
+            history_notes.setdefault(component, []).append(note)
+    # if description has no history notes, parse PR title/commit message
+    # otherwise should skip PR title/commit message
+    if not note_in_desc:
+        component, note = parse_message(lines[0])
+        if component is not None:
+            history_notes.setdefault(component, []).append(note)
```

Comment:
Any candidate ways instead of skip it? I am not sure whether it's fine.

Reply:
It depends on whether we follow the [rule](https://github.com/Azure/azure-cli/blob/9444040ac8435f291d38e95244aa5e97d40205db/doc/authoring_command_modules/README.md#format-pr-description). If we write multiple history notes in PR description, then PR title needs to be a summary, not a history note.

If there's only one history note writen in PR description, I think this process should also be fine. In a word, history notes in PR decription will overwrite the one in PR title.